File: C++/maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int n1 = nums1.size();
        int n2 = nums2.size();
        int dist = 0;
        int i = 0, j = 0;
        while (i < n1 && j < n2) {
            if (nums1[i] <= nums2[j]) {
                dist = max(dist, j - i);
                j++;
            } else if (i < j) {
                i++;
            } else {
                j++;
            }
        }
        return dist;
    }
};

File: JavaScript/maximum-distance-between-a-pair-of-values.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var maxDistance = function(nums1, nums2) {
    let i = 0
    let j = 0
    let dist = 0
    const n1 = nums1.length
    const n2 = nums2.length
    while (i < n1 && j < n2) {
        if (nums1[i] <= nums2[j]) {
            dist = Math.max(dist, j - i)
            j++
        } else if (i < j) {
            i++
        } else {
            j++
        }
    }
    return dist
};